### ISSUE TITLE
Custom Registry Cafile

### ISSUE THREAD

**Issue by sprengerst** [BePo65/license-report#41]
Hi,

thanks for your amazing work.

I saw that your license-report config serves the ability to override the npm registry.
Is there also the ability to add a **cafile**? Otherwise if you are behind a proxy, the custom registry is not practicable.

```
http request to npm for package "sharp" failed, retrying again soon...
Error: unable to get local issuer certificate
    at TLSSocket.onConnectSecure (_tls_wrap.js:1497:34)
    at TLSSocket.emit (events.js:314:20)
    at TLSSocket._finishInit (_tls_wrap.js:932:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:706:12) {
  code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY'
}
```

Best,
Stefan

**Comment by BePo65** [comment]
License-Report uses the npm package 'request' to access the registry.
From the documentation of 'request' I see that there are some environment variables respected for proxies (see ['request'](https://www.npmjs.com/package/request#proxies) documentation on proxies for details).

Will this solve your issue? Sorry, but I don't have a proxied registry at hand to test it on my own.

**Comment by sprengerst** [comment]
Hi BePo,
I will give it a try. I will comment if this is not solved the issue.
Thanks for your guidance.

**Comment by BePo65** [comment]
I even found an example in stackoverflow: [request with proxy](https://stackoverflow.com/questions/48973288/how-to-use-proxy-in-npms-request-package).

**Comment by paddy-hack** [comment]
For those coming here, the use of `request` has been replaced with `got` and proxy support is no longer present 😢

**Comment by BePo65** [comment]
@paddy-hack  Seems that we have lost this feature while improving license-report, but I admit that proxy is a real topic especially in corporate environments.

So I have a few points / questions:

1. I suppose you are looking for a solution to one of your own use cases. If so, perhaps you should create a new issue with this feature request
2. `got` has a few ideas concerning [proxies](https://github.com/sindresorhus/got/blob/main/documentation/tips.md#proxying). Do you have used any of these before? Can you give me a recommendation?
3. Can you help me find a solution to simulate a proxy for testing purposes (e.g. a npm package like nock or a docker image)?
4. Do you have the time for a pull request?

**Comment by paddy-hack** [comment]
> 1. I suppose you are looking for a solution to one of your own use cases.

Indeed, I was.  I have been looking at several tools to help me generate a list of licenses of the (recursive) `dependencies` for the frontend part of an application maintained in a monorepo, i.e. there are multiple workspaces.  Nothing I looked at worked satisfactorily so I have rolled my own, using `package-lock.json` as input.  That requires no network access and is blazingly fast.  Of course, this assumes that `package-lock.json` is correct and all packages list their license.

As to your other questions, the answer is unfortunately "no" :cry:
